R/analyse_pseudo_bulk.R

```diff
@@ -0,0 +1,331 @@
+setGeneric("generate_pseudo_bulk_data", function(object, 
+                                                 group_by = NULL, 
+                                                 split_by  = "random", 
+                                                 k_variable = "4") standardGeneric("generate_pseudo_bulk_data"))
+
+#' generate_pseudo_bulk_data using Seurat object
+#' @inherit generate_pseudo_bul_data 
+#' @param group_by      entry in metadata table, based on these cluster annotation pseudo bulk is performed
+#' @param split_by      variable -> split by a variable within the metadata; k must be a string
+#'                      random -> splits based on a random number; k must be a number
+#'                      Louvain, Louvain_multilevel, SLM, Leiden -> subclusters k must be a list with [resolution, cluster_1, cluster_2]
+#' @param k_variable    variable dependent on the split_by
+#'
+#' @returns             returns pseudo bulk generated data
+#' @examples
+#' random <- generate_pseudo_bulk_data(test_data,"seurat_clusters","random",2)
+#' variable <- generate_pseudo_bulk_data(test_data,"seurat_clusters","random","samples_id")
+#' louvain_clustering <- generate_pseudo_bulk_data(test_data,"seurat_clusters","Louvian",list(4,0,1)
+#'
+#' @export
+setMethod("generate_pseudo_bulk_data", c("object" = "Seurat"), function(object, 
+                                                                        group_by, 
+                                                                        split_by = "random", 
+                                                                        k_variable){
+  
+  if (split_by == "variable"){
```

Comment:
For each of these cases, I suggest you add a check that the `k_variable` is as expected (ie a string, etc). If not, display an error message using the `stop` function which also exits the function.